src/payload.rs

```diff
@@ -458,6 +464,20 @@ where
         self.grant.commit(payload_len + EsbHeader::header_size())
     }
 
+    /// Set the amount to automatically commit on drop
+    ///
+    /// If `None` is given, then the packet will not be commited. If `Some(0)`
+    /// is given, then an empty packet will be committed automatically
+    pub fn to_commit(&mut self, amt: Option<usize>) {
+        if let Some(amt) = amt {
+            let payload_len = self.payload_len().min(amt);
```

Comment:
Hmm. This will only allow us to ever shrink the auto commit amount. I think we could instead take `(self.grant.len() - EsbHeader::header_size()).min(amt)`, which might be more correct? Any thoughts?

Reply:
Hmm, I guess it makes sense, maybe we should do the same for the `update_header` method.

Reply:
Made this change in both places. I also used saturating sub, for defensiveness.